backend/src/main/java/fk/prof/backend/model/request/RecordedProfileProcessor.java

```diff
@@ -0,0 +1,124 @@
+package fk.prof.backend.model.request;
+
+import fk.prof.backend.aggregator.AggregationWindow;
+import fk.prof.backend.exception.AggregationFailure;
+import fk.prof.backend.service.IProfileWorkService;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import recording.Recorder;
+
+import java.io.IOException;
+import java.time.Clock;
+import java.time.LocalDateTime;
+
+public class RecordedProfileProcessor {
+  private static Logger logger = LoggerFactory.getLogger(RecordedProfileProcessor.class);
+
+  private IProfileWorkService profileWorkService;
+  private ISingleProcessingOfProfileGate singleProcessingOfProfileGate;
+
+  private RecordedProfileHeader header = null;
+  private AggregationWindow aggregationWindow = null;
+  private long workId = 0;
+  private LocalDateTime startedAt = null;
+
+  private RecordedProfileHeaderParser headerParser;
+  private WseParser wseParser;
+  private RecordedProfileIndexes indexes = new RecordedProfileIndexes();
+
+  private boolean intermediateWseEntry = false;
+
+  public RecordedProfileProcessor(IProfileWorkService profileWorkService, ISingleProcessingOfProfileGate singleProcessingOfProfileGate,
+                                  int maxAllowedBytesForRecordingHeader, int maxAllowedBytesForWse) {
+    this.profileWorkService = profileWorkService;
+    this.singleProcessingOfProfileGate = singleProcessingOfProfileGate;
+    this.headerParser = new RecordedProfileHeaderParser(maxAllowedBytesForRecordingHeader);
+    this.wseParser = new WseParser(maxAllowedBytesForWse);
+  }
+
+  /**
+   * Returns true if header has been successfully parsed and no wse entry log is parsed partially
+   *
+   * @return parsed a valid recorded profile object or not
+   */
+  public boolean isParsed() {
+    return headerParser.isParsed() && !intermediateWseEntry;
+  }
+
+  /**
+   * Reads buffer and updates internal state with parsed fields.
+   * Aggregates the parsed entries in appropriate aggregation window
+   * Returns the starting unread position in outputstream
+   *
+   * @param inputStream
+   */
+  public void process(CompositeByteBufInputStream inputStream) throws AggregationFailure {
+    if (startedAt == null) {
+      startedAt = LocalDateTime.now(Clock.systemUTC());
+    }
+
+    try {
+      if (!headerParser.isParsed()) {
```

Comment:
Replacing this with aggregationWindow == null check which is functionally more relevant